tools/tesseratos/src/tesseratos/importer/plugin.cpp

```diff
@@ -0,0 +1,303 @@
+        paletteHandle = assets.create(VoxelPalette{});
+        assets.writeMeta(paletteHandle)->set("path", state.paletteFilePath);
+        assets.save(paletteHandle);
+    }
+
+    std::vector<AnyAsset> gridHandles;
+    std::vector<std::string> gridPaths;
+    gridHandles.resize(state.gridCount);
+    gridPaths.resize(state.gridCount);
+    for (std::size_t i = 0; i < state.gridCount; ++i)
+    {
+        if (state.gridFilePaths.contains(i))
+        {
+            gridPaths[i] = state.gridFilePaths.at(i);
+            gridHandles[i] = assets.find(gridPaths[i]);
+        }
+    }
+
+    {
+        // Read the current VoxelModel asset.
+        auto model = assets.read<VoxelModel>(state.currentAsset);
+
+        // Write to the palette asset.
+        auto palette = assets.write<VoxelPalette>(paletteHandle);
+
+        // Iterate over the grids which will be exported.
+        for (std::size_t i = 0; i < state.gridCount; ++i)
+        {
+            if (state.gridFilePaths.contains(i))
+            {
+                if (gridHandles[i].isNull())
+                {
+                    CUBOS_INFO("Failed to get asset handle for gridPath: {}", gridPaths[i]);
+                    CUBOS_INFO("Creating new grid asset.");
+
+                    // Create a new empty VoxelGrid and assign a handle to it.
+                    gridHandles[i] = assets.create(VoxelGrid{});
+                    assets.writeMeta(gridHandles[i])->set("path", gridPaths[i]);
+                    assets.save(gridHandles[i]);
+                }
+
+                // Merge the current model's palette into the main palette.
+                palette.get().merge(model.get().palette(), 1.0F);
+
+                {
+                    auto grid = assets.write<VoxelGrid>(gridHandles[i]);
+
+                    // Convert the grid using the main palette.
+                    if (!grid.get().convert(model.get().palette(), palette.get(), 1.0F))
+                    {
+                        CUBOS_ERROR("Error converting grid to main palette");
+                        return false;
+                    }
+                } // Release lock on gridHandle in order to save it.
+
+                // Save the grid to the given path.
+                CUBOS_INFO("Saving grid to path: {}", gridPaths[i]);
+                if (!assets.save(gridHandles[i]))
+                {
+                    CUBOS_ERROR("Error saving grid to path: {}", gridPaths[i]);
+                    return false;
+                }
+            }
+        }
+    } // Release lock on paletteHandle in order to save it.
+
+    // Save the palette to the asset system.
+    CUBOS_INFO("Saving palette to path: {}", state.paletteFilePath);
+    if (!assets.save(paletteHandle))
+    {
+        CUBOS_ERROR("Error saving palette to asset system");
+        return false;
+    }
+
+    assets.writeMeta(state.currentAsset)->set("palette_id", uuids::to_string(paletteHandle.getId()));
```

Comment:
```suggestion
    assets.writeMeta(state.currentAsset)->set("paletteId", uuids::to_string(paletteHandle.getId()));
```
Just for consistency with the other json files we have